receiver/mongodbatlasreceiver/log_decode_test.go

```diff
@@ -226,3 +226,266 @@ func TestDecode5_0NotGzip(t *testing.T) {
 func strp(s string) *string {
 	return &s
 }
+
+func TestDecodeAudit4_2(t *testing.T) {
+	b, err := os.ReadFile(filepath.Join("testdata", "logs", "sample-payloads", "4.2_audit.log"))
+	require.NoError(t, err)
+
+	zippedBuffer := &bytes.Buffer{}
+	gzipWriter := gzip.NewWriter(zippedBuffer)
+
+	_, err = gzipWriter.Write(b)
+	require.NoError(t, err)
+	require.NoError(t, gzipWriter.Close())
+
+	entries, err := decodeAuditJSON(zippedBuffer)
+	require.NoError(t, err)
+
+	t.Logf("%#v", entries)
+
+	require.Equal(t, []model.AuditLog{
```

Comment:
These decoded audit logs don't contain all the information they should; See issue #14177 - these tests do check that they unmarshal successfully in the normal case, and fail out without infinitely looping in the malformed case.